src/python_testing/TC_ECOINFO_2_2.py

```diff
@@ -0,0 +1,118 @@
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body
+from mobly import asserts
+
+
+class TC_ECOINFO_2_2(MatterBaseTest):
+
+    def steps_TC_ECOINFO_2_2(self) -> list[TestStep]:
+        steps = [TestStep(1, "Prepare", is_commissioning=True),
+                 TestStep("1a", "Read root endpoint's PartsList"),
+                 TestStep("1b", "For each endpoint in 1a read DeviceType list confirming aggregator endpoint exists"),
+                 TestStep(2, "Add a bridged device"),
+                 TestStep("2a", "(Manual Step) Add a bridged device using method indicated by the manufacturer"),
+                 TestStep("2b", "Read root endpoint's PartsList, validate exactly one endpoint added"),
+                 TestStep("2c", "On newly added endpoint detected in 2b read RemovedOn Ecosystem Information Attribute and validate"),
+                 TestStep(3, "Remove bridged device"),
+                 TestStep("3a", "(Manual Step) Removed bridged device added in step 2a using method indicated by the manufacturer"),
+                 TestStep("3b", "On newly added endpoint detected in 2b read RemovedOn Ecosystem Information Attribute and validate"),
+                 TestStep("3c", "On newly added endpoint detected in 2b read DeviceDirectory Ecosystem Information Attribute and validate"),
+                 TestStep("3d", "On newly added endpoint detected in 2b read LocationDirectory Ecosystem Information Attribute and validate")]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_ECOINFO_2_2(self):
+        dev_ctrl = self.default_controller
+        dut_node_id = self.dut_node_id
+
+        self.print_step(0, "Commissioning, already done")
+        self.step(1)
+        self.step("1a")
+        root_node_endpoint = 0
+        root_part_list = await dev_ctrl.ReadAttribute(dut_node_id, [(root_node_endpoint, Clusters.Descriptor.Attributes.PartsList)])
+
+        self.step("1b")
+        set_of_endpoints_step_1 = set(root_part_list[root_node_endpoint]
+                                      [Clusters.Descriptor][Clusters.Descriptor.Attributes.PartsList])
+        list_of_aggregator_endpoints = []
+        for endpoint in set_of_endpoints_step_1:
+            device_type_list_read = await dev_ctrl.ReadAttribute(dut_node_id, [(endpoint, Clusters.Descriptor.Attributes.DeviceTypeList)])
+            device_type_list = device_type_list_read[endpoint][Clusters.Descriptor][Clusters.Descriptor.Attributes.DeviceTypeList]
+            for device_type in device_type_list:
+                if device_type.deviceType == 14:
```

Comment:
add a named constant for `14`? like DEVICE_TYPE_AGGREGGATOR (and maybe use 0x000E to match spec formatting for constant)